lib/generators/boring/pronto/gitlab_ci/install/install_generator.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require "generators/boring/pronto/base_generator"
+require "boring_generators/generator_helper"
+
+module Boring
+  module Pronto
+    module GitlabCi
+      class InstallGenerator < Boring::Pronto::BaseGenerator
+        desc "Adds pronto gem with various extensions and configure for Gitlab CI"
+        source_root File.expand_path("templates", __dir__)
+
+        class_option :ruby_version, type: :string, aliases: "-rv"
+
+        include BoringGenerators::GeneratorHelper
+
+        def add_configuration_to_gitlab_ci_file
+          say "Adding Pronto configurations to .gitlab-ci.yml", :green
+
+          if gitlab_ci_file_content["pronto"]
+            say "Pronto configurations already exist! Skipping them ...", :yellow
+
+            return
+          end
+
+          ruby_version = options.ruby_version || app_ruby_version
+
+          ci_file_content = <<~RUBY
```

Comment:
can we create a template for this? based on the class_options flag we can set the instance variables and template them in one go.

For example: 
```
@insert_stages = ci_file_content["stages"]

## Template file
<% if @insert_stages %>
stages:
  - lint
<% end%>
```

To goal is to keep this code clean and templatize the `gitlab-ci.yml` based on the class option flags. This would ensure we return correct templatized file as output.